doc/source/development/contributing_environment.rst

```diff
@@ -204,9 +244,22 @@ At this point you should be able to import pandas from your locally built versio
    >>> print(pandas.__version__)  # note: the exact output may differ
    2.0.0.dev0+880.g2b9e661fbb.dirty
 
-This will create the new environment, and not touch any of your existing environments,
-nor any existing Python installation.
-
-.. note::
-   You will need to repeat this step each time the C extensions change, for example
-   if you modified any file in ``pandas/_libs`` or if you did a fetch and merge from ``upstream/main``.
+When building pandas with meson, importing pandas will automatically trigger a rebuild, even when C/Cython files are modified. 
```

Comment:
even when no C/Cython files are modified? (I think you are missing a no here)

Does this mean every time I am running a script that imports pandas through an IDE this will trigger a build?

Reply:
It actually needs to rebuild every time unfortunately.

Rebuilding will never be a no-op, since we need to check if the commit changed and re-write the version file.
(This was why the setuptools editable install was slow to import also, since versioneer is slow)

Also, meson needs to check if it needs to sync files between the source directory and the build directory every time, which is the price we pay for a clean source directory.

Right now, I think meson is ~1 second slower(it copies all files even if they are not changed, which is something I'm going to optimize) than setuptools, but I expect it to take about the same time by the time we remove setuptools.

Reply:
Ah got you, I misunderstood the docs then. I thought we actually have to compile everything all the time (rebuild sounded like that for me). 1 second isn't too bad